fastplotlib/widgets/image.py

```diff
@@ -368,13 +368,21 @@ def __init__(
             if all([_is_arraylike(d) for d in data]):
                 # Grid computations
                 if figure_shape is None:
-                    figure_shape = calculate_figure_shape(len(data))
+                    if figure_kwargs is not None:
+                        if "shape" in figure_kwargs:
+                            figure_shape = figure_kwargs["shape"]
+                    else:
+                        figure_shape = calculate_figure_shape(len(data))
```

Comment:
what about initialize `figure_kwargs` before here instead of later (see L511), so then we can just do this and remove the one if block:

```python
if figure_shape is None:
  if "shape" in figure_kwargs.keys()():
    figure_shape = figure_kwargs["shape"]
...
```

Reply:
Agreed! Just updated this in the PR